misk-redis/src/main/kotlin/misk/redis/RedisService.kt

```diff
@@ -0,0 +1,24 @@
+package misk.redis
+
+import com.google.common.util.concurrent.AbstractIdleService
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+/**
+ * Controls the connection lifecycle for Redis.
+ */
+@Singleton
+internal class RedisService @Inject internal constructor(
+  private val redisProvider: Provider<Redis>
+) : AbstractIdleService() {
+  private lateinit var redis: Redis
+
+  override fun startUp() {
+    redis = redisProvider.get()
```

Comment:
Why do we need to lazy init the `redis` instance?

Reply:
it connects to the instance on the Constructor

Reply:
Yeah, instantiating `Jedis` connects to the Redis instance so we can't initialize the property as we declare it, adding a comment to clarify